elastic/kibana
# [Maps] fallback to GeoJSON with runtime geo_point field

Fixes https://github.com/elastic/kibana/issues/136424

Vector tile search API does not support runtime geo_point fields when searching for hits because [value is returned as lat,lon string instead of GeoJSON Point](https://github.com/elastic/elasticsearch/issues/85245)

This PR updates Documents source to fallback to GeoJSON when geospatial field is runtime field. Vector tiles option is disabled in the scaling form.

Note: cluster source works with runtime geo_point field and vector tiles.

<img width="600" alt="Screen Shot 2022-07-19 at 12 55 52 PM" src="https://user-images.githubusercontent.com/373691/179832497-c13a3caa-a5ee-431e-83d8-360f9a81dd17.png">

To test:
* Create a simple index with lat/lon data
    ```
    DELETE runtime_geo_example

    PUT runtime_geo_example
    {
      "mappings": {
        "properties": {
          "latitude": {"type": "float"},
          "longitude": {"type": "float"}
        }
      }
    }

    PUT runtime_geo_example/_bulk
    { "index": {} }
    { "longitude": 0.5, "latitude": 40}
    { "index": {} }
    { "longitude": 0, "latitude": 41}
    { "index": {} }
    { "longitude": -0.5, "latitude": 39}

    ```
* Create data view for `runtime_geo_example` index
* Add geo_point runtime field to data view. Set value to `emit(doc['longitude'].value,doc['latitude'].value)`
* Create a new map.
* Add a document layer and select `runtime_geo_example` data view. Verify results are displayed and "Vector tile" scaling option is disabled.
